controllers/common.go

```diff
@@ -439,3 +439,67 @@ func (r *SearchReconciler) createSecret(ctx context.Context, secret *corev1.Secr
 func DeploymentEquals(current, new *appsv1.Deployment) bool {
 	return equality.Semantic.DeepEqual(current.Spec, new.Spec)
 }
+
+// update status condition in search instance
+func updateStatusCondition(instance *searchv1alpha1.Search, podList *corev1.PodList) *searchv1alpha1.Search {
+	var podCondition metav1.Condition
+	var readyType string
+
+	for _, searchPod := range podList.Items {
```

Comment:
Will this logic work when we add multiple replicas of these pods? Should we use the deployment's `MinimumReplicasAvailable` condition instead?

Reply:
It will work for Multiple replicas of ,say search-api pods. I will check all pods status and if any one of them is not ready the status will have the failure information. We thought about using deployments replica count , but then we had to again find the pod and get the cause of the failure.

Reply:
thanks!